src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -416,15 +416,18 @@ public List<int> ConsensusTipChanged(ChainedHeader newConsensusTip, bool blockMi
         {
             ChainedHeader oldConsensusTip = this.GetConsensusTip();
             ChainedHeader fork = newConsensusTip.FindFork(oldConsensusTip);
+            ChainedHeader currentHeader = newConsensusTip;
 
             this.logger.LogTrace("Old consensus tip: '{0}', new consensus tip: '{1}', fork point: '{2}'.", oldConsensusTip, newConsensusTip, fork);
 
-            if (!blockMined && newConsensusTip.Block != null)
+            // Consider blocks that became a part of our best chain as consumed.
+            while ((currentHeader != fork) && !blockMined && (newConsensusTip.Block != null))
```

Comment:
`newConsensusTip.Block != null`

Is this even possible?

Reply:
yeah. it's null at initialization (we call setCT when we initialize CM)

Reply:
ah I suggest a small change here to put the `!blockMined && (newConsensusTip.Block != null)` in an `if` before the loop even starts.
do it in another PR its ok.

Reply:
The loop won't start on initialize at all because of that if block != null

Reply:
Yeah but you will hit those conditions  on all other cases

Reply:
Ah, you mean just move the condition of block being bot null in front? Sure, 1 sec

Reply:
fixed